src/api/app/controllers/cloud/upload_jobs_controller.rb

```diff
@@ -3,17 +3,14 @@ class UploadJobsController < ApplicationController
     before_action :require_login
     before_action -> { feature_active?(:cloud_upload) }
     before_action :validate_configuration_presence, only: [:index, :create]
+    before_action :set_upload_job, only: [:destroy, :show]
 
     def index
       render xml: ::Cloud::Backend::UploadJob.all(::User.current, format: :xml)
     end
 
     def show
-      if ::User.current.upload_jobs.where(job_id: params[:id]).exists?
-        render xml: ::Cloud::Backend::UploadJob.find(params[:id], format: :xml)
-      else
-        render_error status: 404
-      end
+      render xml: ::Cloud::Backend::UploadJob.find(params[:id], format: :xml)
```

Comment:
``
render xml: ::Cloud::Backend::UploadJob.find(@upload_job.job_id, format: :xml)
`` 

because it looked at the first glance that there is no access check (which happens in the before filter)